thundra/plugins/trace/patcher.py

```diff
@@ -1,95 +1,126 @@
+                    if module_path != None:
+                        modules[module_path] = [function_prefix, arguments]
+
         return modules
-    
-    def get_module_function_prefix(self, module_name):
-        try: 
-            return self.modules_map[module_name][0]
-        except:
-            return None
-    
-    def get_trace_arguments(self, module_name):
-        try: 
-            return self.modules_map[module_name][1]
-        except:
-            return None
 
 
 class ThundraFinder(PathFinder):
 
-    def __init__(self, module_name):
-        self.module_name = module_name
+    def __init__(self, module_path_regex, function_prefix, trace_arguments):
+        self.module_path_regex = module_path_regex
+        self.function_prefix = function_prefix
+        self.trace_arguments = trace_arguments
 
     def find_spec(self, fullname, path=None, target=None):
-        if fullname == self.module_name:
-            spec = super().find_spec(fullname, path, target)
-            loader = ThundraLoader(fullname, spec.origin)
-            return ModuleSpec(fullname, loader)
-
+        try:
+            if (re.match(self.module_path_regex, fullname)):
+                spec = super().find_spec(fullname, path, target)
+                if not spec:
+                    return None
+                loader = ThundraLoader(fullname, spec.origin, self.function_prefix, self.trace_arguments)
+                _spec = ModuleSpec(fullname, loader)
+                _spec.submodule_search_locations = spec.submodule_search_locations
+                return _spec
+        except Exception as e:
+            print(e)
+            return None
+    
 
 # Loading the module in a load time
 class ThundraLoader(SourceFileLoader):
-  
+    def __init__(self, fullname, path, function_prefix, trace_args):
+        super().__init__(fullname, path)
+        self.function_prefix = function_prefix
+        self.trace_args = trace_args
+
     def exec_module(self, module):
-        super().exec_module(module)
-        import_patcher = ImportPatcher()
-        module_name = utils.get_module_name(module)
-        function_prefix = import_patcher.get_module_function_prefix(module_name)
-        trace_args_list = import_patcher.get_trace_arguments(module_name)
         try:
-            trace_args = utils.str2bool(trace_args_list[constants.TRACE_ARGS])
+            super().exec_module(module)
+        except Exception as e:
+            print(e)
+            return module
+
+        try:
+            import_patcher = ImportPatcher()
+            module_name = utils.get_module_name(module)
+
+            trace_args, trace_return_value, trace_error = self.get_trace_parameters()
+            allowed_functions = utils.get_allowed_functions(module)
+
+            if self.function_prefix != None:
+                for function in allowed_functions:
+                    if (self.function_prefix == "*") or (self.function_prefix in function['functionName']):
```

Comment:
`...or (self.function_prefix in function['functionName'])` this line cause a function with the name `testf` to match configuration `f*`. Instead of using in operator we can use regex match.